H2MLauncher.Core/Services/H2MLauncherService.cs

```diff
@@ -1,21 +1,44 @@
-﻿using System.Text.Json;
+﻿using System.Diagnostics;
+using System.Text.Json;
+
+using H2MLauncher.Core.Utilities;
 
 namespace H2MLauncher.Core.Services
 {
-    public class H2MLauncherService(HttpClient httpClient, IErrorHandlingService errorHandlingService)
+    public class H2MLauncherService
     {
         private const string GITHUB_REPOSITORY = "https://api.github.com/repos/Bowhza/H2M-Launcher/releases";
-        private readonly HttpClient _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
-        private readonly IErrorHandlingService _errorHandlingService = errorHandlingService ?? throw new ArgumentNullException(nameof(errorHandlingService));
-        public const string CURRENT_VERSION = "H2M-v2.0.2";
+        private const string LAUNCHER = "H2MLauncher.UI.exe";
+        private const string LAUNCHER_BACKUP = $"{LAUNCHER}.backup";
+        private readonly HttpClient _httpClient;
+        private readonly IErrorHandlingService _errorHandlingService;
+        
+        public const string CURRENT_VERSION = "H2M-v2.0.3";
```

Comment:
Can we somehow get rid of this constant? We should really pull the version from AssemblyInfo as a single source of truth. (i.e. `Assembly.GetExecutingAssembly().GetName().Version`). Maybe we can craft this string in the ViewModel based on that?